home/pi/auto_run.sh

```diff
@@ -548,6 +566,85 @@ then
     rm ~/first_run
 fi
 
+# Matrix Voice Hat Setup
+if [ -f ~/setup_matrix ]
+then
+    if [ ! -f matrix_setup_state.txt ]
+    then
+        echo ""
+        echo "========================================================================="
+        echo "Setting up Matrix Voice Hat. This will install the matrixio-kernel-modules and pulseaudio"
+        echo "This will require 3 reboots"
+        echo "Press any key to continue..."
+        read -N1 -s anykey
+    else
+        echo "Press any key to continue setting up Matrix Voice HAT"
+        read -N1 -s anykey
+    fi
+
+    if [ ! -f matrix_setup_state.txt ]
+    then
+        echo "Add Matrix repo and key and Update and install packages"
```

Comment:
echo "Adding Matrix repo and installing packages..."

No need for all that detail.